File: python/61c_1497A_Meximization.py
def solve(arr, n):
    ans = []
    arr.sort()
    for i in range(n):
        if arr[i] != arr[i-1]:
            ans.append(arr[i])
    for i in range(n):
        if arr[i] == arr[i-1]:
            ans.append(arr[i])
    return ans

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        ans.append(solve(arr, n))
    for i in ans:
        for j in i:
            print(j, end = " ")
        print()

if __name__ == "__main__":
    main()

File: cpp/61c_1497A_Meximization.cpp
#include<bits/stdc++.h>

using namespace std;

vector<int> solve(vector<int> arr, int n){
    vector<int> ans;
    sort(arr.begin(), arr.end());
    for(int i = 0; i < n; i++){
        if(arr[i] != arr[i-1]){
            ans.push_back(arr[i]);
        }
    }
    for(int i = 0; i < n; i++){
        if(arr[i] == arr[i-1]){
            ans.push_back(arr[i]);
        }
    }
    return ans;
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, temp;
    vector<vector<int>> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n;
        vector<int> arr;
        for(int j = 0; j < n; j++){
            cin>>temp;
            arr.push_back(temp);
        }
        ans.push_back(solve(arr, n));
    }

    for(auto i : ans){
        for(auto j : i){
            cout<<j<<" ";
        }
        cout<<"\n";
    }
    return 0;
}